conjure-api/conjure/spec/enums.conjure.go

```diff
@@ -13,167 +13,239 @@ import (
 
 var enumValuePattern = regexp.MustCompile("^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$")
 
-type ErrorCode string
+type ErrorCode struct {
+	val ErrorCodeValue
+}
+
+type ErrorCodeValue string
 
 const (
-	ErrorCodePermissionDenied      ErrorCode = "PERMISSION_DENIED"
-	ErrorCodeInvalidArgument       ErrorCode = "INVALID_ARGUMENT"
-	ErrorCodeNotFound              ErrorCode = "NOT_FOUND"
-	ErrorCodeConflict              ErrorCode = "CONFLICT"
-	ErrorCodeRequestEntityTooLarge ErrorCode = "REQUEST_ENTITY_TOO_LARGE"
-	ErrorCodeFailedPrecondition    ErrorCode = "FAILED_PRECONDITION"
-	ErrorCodeInternal              ErrorCode = "INTERNAL"
-	ErrorCodeTimeout               ErrorCode = "TIMEOUT"
-	ErrorCodeCustomClient          ErrorCode = "CUSTOM_CLIENT"
-	ErrorCodeCustomServer          ErrorCode = "CUSTOM_SERVER"
+	ErrorCodePermissionDenied      ErrorCodeValue = "PERMISSION_DENIED"
+	ErrorCodeInvalidArgument       ErrorCodeValue = "INVALID_ARGUMENT"
+	ErrorCodeNotFound              ErrorCodeValue = "NOT_FOUND"
+	ErrorCodeConflict              ErrorCodeValue = "CONFLICT"
+	ErrorCodeRequestEntityTooLarge ErrorCodeValue = "REQUEST_ENTITY_TOO_LARGE"
+	ErrorCodeFailedPrecondition    ErrorCodeValue = "FAILED_PRECONDITION"
+	ErrorCodeInternal              ErrorCodeValue = "INTERNAL"
+	ErrorCodeTimeout               ErrorCodeValue = "TIMEOUT"
+	ErrorCodeCustomClient          ErrorCodeValue = "CUSTOM_CLIENT"
+	ErrorCodeCustomServer          ErrorCodeValue = "CUSTOM_SERVER"
+	ErrorCodeUnknown               ErrorCodeValue = "UNKNOWN"
 )
 
 // ErrorCode_Values returns all known variants of ErrorCode.
-func ErrorCode_Values() []ErrorCode {
-	return []ErrorCode{ErrorCodePermissionDenied, ErrorCodeInvalidArgument, ErrorCodeNotFound, ErrorCodeConflict, ErrorCodeRequestEntityTooLarge, ErrorCodeFailedPrecondition, ErrorCodeInternal, ErrorCodeTimeout, ErrorCodeCustomClient, ErrorCodeCustomServer}
+func ErrorCode_Values() []ErrorCodeValue {
+	return []ErrorCodeValue{ErrorCodePermissionDenied, ErrorCodeInvalidArgument, ErrorCodeNotFound, ErrorCodeConflict, ErrorCodeRequestEntityTooLarge, ErrorCodeFailedPrecondition, ErrorCodeInternal, ErrorCodeTimeout, ErrorCodeCustomClient, ErrorCodeCustomServer}
+}
+
+func NewErrorCode(value ErrorCodeValue) ErrorCode {
+	return ErrorCode{val: value}
 }
 
 // IsUnknown returns false for all known variants of ErrorCode and true otherwise.
 func (e ErrorCode) IsUnknown() bool {
```

Comment:
Given we're making a break here, does it makes sense to make this method private? I'm fine keeping it, but it seems a little redundant after reinstating `EnumUnknown` values. Just a thought.

Reply:
I still think `if v.IsUnknown() {` is cleaner than `if v.Value() == api.EnumUnknown` so would prefer to keep it